src/lib/source-handlers/bitbucket-server/list-projects.ts

```diff
@@ -0,0 +1,87 @@
+import * as needle from 'needle';
+import * as debugLib from 'debug';
+import Bottleneck from 'bottleneck';
+import { BitbucketServerProjectData } from './types';
+import { getBitbucketServerToken } from './get-bitbucket-server-token';
+
+const debug = debugLib('snyk:bitbucket-server');
+
+const limiter = new Bottleneck({
+  reservoir: 1000, // initial value
+  reservoirRefreshAmount: 1000,
+  reservoirRefreshInterval: 3600 * 1000,
+  maxConcurrent: 1,
+  minTime: 1000,
+});
+
+limiter.on('failed', async (error, jobInfo) => {
+  const id = jobInfo.options.id;
+  console.warn(`Job ${id} failed: ${error}`);
+  if (jobInfo.retryCount === 0) {
+    // Here we only retry once
+    console.log(`Retrying job ${id} in 25ms!`);
+    return 25;
+  }
+});
+
+export const fetchAllProjects = async (
+  url: string,
+  token: string,
+): Promise<unknown[]> => {
```

Comment:
not unknown it's `BitbucketServerProjectData[]` thats why you have to cast later, make sure to return the relevant type in the functions.

Reply:
done